lib/shopify_api/collection_pagination.rb

```diff
@@ -0,0 +1,52 @@
+module ShopifyAPI
+  module CollectionPagination
+
+    def next_page?
+      next_page_info.present?
+    end
+
+    def previous_page?
+      previous_page_info.present?
+    end
+
+    def fetch_next_page
+      fetch_page(next_page_info)
+    end
+
+    def fetch_previous_page
+      fetch_page(previous_page_info)
+    end
+
+    private
+
+    AVAILABLE_IN_VERSION = ShopifyAPI::ApiVersion::Unstable.new
+
+    def fetch_page(page_info)
+      return [] unless page_info
+
+      resource_class.where(original_params.merge(page_info: page_info))
```

Comment:
If your first request had the param `order: 'title asc'` then this would merge page_info to that? Because that would make a bad URL. If we tried to request a page against Shopify with the original params + page_info it will fail if there are filters or sort orders in there.

We're probably better off just parsing out all the params out of the next/previous url and using them directly

Reply:
Discussed offline, I'm going to take that approach, parse the next/previous page and use those for params.